### PR TITLE
Implemented NOT operators and other comparison operators

### PR DESCRIPTION
**Pull request by nelio2k**
- Found and fixed an issue where false JSON values were perceived as true.
- SimpleParser added a NOT operator, which is used in "!=", "<=", and ">=" operations. Modules in between simpleParser and matcher were also changed to allow the logic to pass all the way through.
- Discussed with Brett about changing GreaterThanEq in matcher and associates to just Greater... since <= and >= can be accomplished with inverse of "<" and ">".
- Found and fixed issue on mixed-type numeric comparisons, by converting everything to float64 for comparison. It's not the most efficient, but at least we can address correctness as of now.
- Beefed up simpleParser unit test to test all the changes mentioned above. The scope of the simpleParser unit test should really be testing just the simpleParser, but it's slowly overstepping its boundary into matcher testing. Not a big deal for now but may address it later.

### CODE REVIEW

**Comment by brett19** [comment]
Please do not include Merge Commits in your pull requests.

File: expression.go [hunk 1]
```diff
@@ -81,6 +81,24 @@ func (expr ValueExpr) RootRefs() []FieldExpr {
 	return nil
 }
 
+type NotExpr []Expression
```
Comment (ysui6888) [inline expression.go]: not expression always has only one subexpr. is it necessary to use an array here?

Comment (brett19) [reply expression.go]: @ysui6888 You should take a look at the implementation here which I believe @nelio2k is going to use to superseed his NotExpr implementation:
https://github.com/couchbaselabs/gojsonsm/pull/11/commits/b9c7abf2e9255180077cd67b1222ebf9507f5607

File: fastval.go [hunk 2]
```diff
@@ -170,13 +170,39 @@ func (val FastVal) AsJsonString() (FastVal, error) {
...
+}
+
 func (val FastVal) compareStrings(other FastVal) int {
 	// TODO: Improve string comparisons to avoid casting or converting
 	escVal, _ := val.AsJsonString()
 	escOval, _ := other.AsJsonString()
 	return strings.Compare(string(escVal.sliceData), string(escOval.sliceData))
 }
 
+func (val FastVal) compareNumeric(other FastVal) int {
```
Comment (ysui6888) [inline fastval.go]: this method assumes that val and other are numeric types. better add a comment to emphasize this

Comment (brett19) [reply fastval.go]: Again, I think this portion of the PR will be superseeded by PR #11 which has a more exhaustive implementation of the various FastVal comparisons.

**Comment by brett19** [comment]
Closing as superseeded by #13

**Pull request closed without merging**